--- ../src-base/minecraft/net/minecraft/init/Items.java
+++ ../src-work/minecraft/net/minecraft/init/Items.java
@@ -1,14 +1,6 @@
 package net.minecraft.init;
 
-import net.minecraft.item.Item;
-import net.minecraft.item.ItemArmor;
-import net.minecraft.item.ItemArmorStand;
-import net.minecraft.item.ItemBow;
-import net.minecraft.item.ItemEmptyMap;
-import net.minecraft.item.ItemFishingRod;
-import net.minecraft.item.ItemMap;
-import net.minecraft.item.ItemPotion;
-import net.minecraft.item.ItemShears;
+import net.minecraft.item.*;
 import net.minecraft.util.ResourceLocation;
 
 public class Items
@@ -223,6 +215,7 @@
     public static final Item field_190930_cZ;
     public static final Item field_191525_da;
     public static final Item field_192397_db;
+    public static final Item CHEESE;
 
     private static Item func_179554_a(String p_179554_0_)
     {
@@ -317,7 +310,7 @@
             field_151159_an = func_179554_a("painting");
             field_151153_ao = func_179554_a("golden_apple");
             field_151155_ap = func_179554_a("sign");
-            field_179570_aq = func_179554_a("wooden_door");
+            field_179570_aq = func_179554_a("oak_door");
             field_179569_ar = func_179554_a("spruce_door");
             field_179568_as = func_179554_a("birch_door");
             field_179567_at = func_179554_a("jungle_door");
@@ -456,6 +449,7 @@
             field_190930_cZ = func_179554_a("shulker_shell");
             field_191525_da = func_179554_a("iron_nugget");
             field_192397_db = func_179554_a("knowledge_book");
+            CHEESE = func_179554_a("cheese");
         }
     }
 }
