--- ../src-base/minecraft/net/minecraft/block/BlockBone.java
+++ ../src-work/minecraft/net/minecraft/block/BlockBone.java
@@ -4,10 +4,8 @@
 import net.minecraft.block.material.Material;
 import net.minecraft.creativetab.CreativeTabs;
 
-public class BlockBone extends BlockRotatedPillar
-{
-    public BlockBone()
-    {
+public class BlockBone extends BlockRotatedPillar {
+    public BlockBone() {
         super(Material.field_151576_e, MapColor.field_151658_d);
         this.func_149647_a(CreativeTabs.field_78030_b);
         this.func_149711_c(2.0F);
