--- ../src-base/minecraft/net/minecraft/block/BlockDoubleStoneSlab.java
+++ ../src-work/minecraft/net/minecraft/block/BlockDoubleStoneSlab.java
@@ -6,15 +6,12 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
 
-public class BlockDoubleStoneSlab extends BlockStoneSlab
-{
-    public boolean func_176552_j()
-    {
+public class BlockDoubleStoneSlab extends BlockStoneSlab {
+    public boolean func_176552_j() {
         return true;
     }
 
-    public BlockFaceShape func_193383_a(IBlockAccess p_193383_1_, IBlockState p_193383_2_, BlockPos p_193383_3_, EnumFacing p_193383_4_)
-    {
+    public BlockFaceShape func_193383_a(IBlockAccess p_193383_1_, IBlockState p_193383_2_, BlockPos p_193383_3_, EnumFacing p_193383_4_) {
         return BlockFaceShape.SOLID;
     }
 }
