--- ../src-base/minecraft/net/minecraft/block/BlockMobSpawner.java
+++ ../src-work/minecraft/net/minecraft/block/BlockMobSpawner.java
@@ -1,6 +1,5 @@
 package net.minecraft.block;
 
-import java.util.Random;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Items;
@@ -15,53 +14,48 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BlockMobSpawner extends BlockContainer
-{
-    protected BlockMobSpawner()
-    {
+import java.util.Random;
+
+public class BlockMobSpawner extends BlockContainer {
+    protected BlockMobSpawner() {
         super(Material.field_151576_e);
     }
 
-    public TileEntity func_149915_a(World p_149915_1_, int p_149915_2_)
-    {
+    public TileEntity func_149915_a(World p_149915_1_, int p_149915_2_) {
         return new TileEntityMobSpawner();
     }
 
-    public Item func_180660_a(IBlockState p_180660_1_, Random p_180660_2_, int p_180660_3_)
-    {
+    public Item func_180660_a(IBlockState p_180660_1_, Random p_180660_2_, int p_180660_3_) {
         return Items.field_190931_a;
     }
 
-    public int func_149745_a(Random p_149745_1_)
-    {
+    public int func_149745_a(Random p_149745_1_) {
         return 0;
     }
 
-    public void func_180653_a(World p_180653_1_, BlockPos p_180653_2_, IBlockState p_180653_3_, float p_180653_4_, int p_180653_5_)
-    {
+    public void func_180653_a(World p_180653_1_, BlockPos p_180653_2_, IBlockState p_180653_3_, float p_180653_4_, int p_180653_5_) {
         super.func_180653_a(p_180653_1_, p_180653_2_, p_180653_3_, p_180653_4_, p_180653_5_);
-        int i = 15 + p_180653_1_.field_73012_v.nextInt(15) + p_180653_1_.field_73012_v.nextInt(15);
-        this.func_180637_b(p_180653_1_, p_180653_2_, i);
     }
 
-    public boolean func_149662_c(IBlockState p_149662_1_)
-    {
+    @Override
+    public int getExpDrop(IBlockState state, net.minecraft.world.IBlockAccess world, BlockPos pos, int fortune) {
+        return 15 + RANDOM.nextInt(15) + RANDOM.nextInt(15);
+    }
+
+    public boolean func_149662_c(IBlockState p_149662_1_) {
         return false;
     }
 
-    public EnumBlockRenderType func_149645_b(IBlockState p_149645_1_)
-    {
+    public EnumBlockRenderType func_149645_b(IBlockState p_149645_1_) {
         return EnumBlockRenderType.MODEL;
     }
 
-    public ItemStack func_185473_a(World p_185473_1_, BlockPos p_185473_2_, IBlockState p_185473_3_)
-    {
+    public ItemStack func_185473_a(World p_185473_1_, BlockPos p_185473_2_, IBlockState p_185473_3_) {
         return ItemStack.field_190927_a;
     }
 
     @SideOnly(Side.CLIENT)
-    public BlockRenderLayer func_180664_k()
-    {
+    public BlockRenderLayer func_180664_k() {
         return BlockRenderLayer.CUTOUT;
     }
 }
