--- ../src-base/minecraft/net/minecraft/block/BlockHalfStoneSlabNew.java
+++ ../src-work/minecraft/net/minecraft/block/BlockHalfStoneSlabNew.java
@@ -1,9 +1,7 @@
 package net.minecraft.block;
 
-public class BlockHalfStoneSlabNew extends BlockStoneSlabNew
-{
-    public boolean func_176552_j()
-    {
+public class BlockHalfStoneSlabNew extends BlockStoneSlabNew {
+    public boolean func_176552_j() {
         return false;
     }
 }
