--- ../src-base/minecraft/net/minecraft/block/state/IBlockProperties.java
+++ ../src-work/minecraft/net/minecraft/block/state/IBlockProperties.java
@@ -1,7 +1,5 @@
 package net.minecraft.block.state;
 
-import java.util.List;
-import javax.annotation.Nullable;
 import net.minecraft.block.material.EnumPushReaction;
 import net.minecraft.block.material.MapColor;
 import net.minecraft.block.material.Material;
@@ -20,18 +18,28 @@
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public interface IBlockProperties
-{
+import javax.annotation.Nullable;
+import java.util.List;
+
+public interface IBlockProperties {
     Material func_185904_a();
 
     boolean func_185913_b();
 
     boolean func_189884_a(Entity p_189884_1_);
 
+    @Deprecated
+        //Forge location aware version below
     int func_185891_c();
 
+    int getLightOpacity(IBlockAccess world, BlockPos pos);
+
+    @Deprecated
+        //Forge location aware version below
     int func_185906_d();
 
+    int getLightValue(IBlockAccess world, BlockPos pos);
+
     @SideOnly(Side.CLIENT)
     boolean func_185895_e();
 
@@ -95,8 +103,17 @@
 
     RayTraceResult func_185910_a(World p_185910_1_, BlockPos p_185910_2_, Vec3d p_185910_3_, Vec3d p_185910_4_);
 
+    @Deprecated
+        // Forge: Use isSideSolid(IBlockAccess, BlockPos, EnumFacing.UP) instead
     boolean func_185896_q();
 
+    //Forge added functions
+    boolean doesSideBlockRendering(IBlockAccess world, BlockPos pos, EnumFacing side);
+
+    boolean isSideSolid(IBlockAccess world, BlockPos pos, EnumFacing side);
+
+    boolean doesSideBlockChestOpening(IBlockAccess world, BlockPos pos, EnumFacing side);
+
     Vec3d func_191059_e(IBlockAccess p_191059_1_, BlockPos p_191059_2_);
 
     boolean func_191058_s();
